test/e2e/istio_test.go

```diff
@@ -364,6 +382,26 @@ func getKialiWorkloadHealth(t *testing.T, kialiAPIUrl string, namespace, workloa
 	return &health, nil
 }
 
+// getKialiWorkloadHealthIstioByWorkloadEndpoint gets metrics of workload by /namespaces/<ns>/workloads/<workload> API.
+// Used in Istio 1.18 and later. Istio 1.17 does not have this API.
+func getKialiWorkloadHealthIstioByWorkloadEndpoint(t *testing.T, kialiAPIUrl string, namespace, workloadName string) (*workloadHealth, error) {
```

Comment:
```suggestion
func getKialiWorkloadHealthIstioByWorkloadEndpoint(t *testing.T, kialiAPIUrl string, namespace, workloadName string) *workloadHealth {
```

Reply:
This changes the signature. I think more changes should be commited for this.

Reply:
yup, I haven't made them because it's not handy to do it from GH